kinematics/acceleration.py

```diff
@@ -0,0 +1,129 @@
+# KINEMATICS #
+
+# Kinematics is the physics of motion
+
+# Here is an explanation on the "big 4 equations", all of which were used to make this
+# algorithm: https://www.physicsclassroom.com/class/1dkin/Lesson-6/Kinematic-Equations
+
+# Output will always be a float, so if you need to round, set retaining_wall to
+# whatever place you are rounding to
+
+# NOTE: Not all of these equations will give you an exactly same answer, but will be
+# very close. Typically you would only show sig
+# figs when you calculate with these equations.
+
+# Doctests are all tested with initial_velocity=0, final_velocity=104.96,
+# initial_position=0, final_position=1720, acceleration=3.2,
+# time_elapsed=32.8, retaining_wall=2
+
+# These were calculated by hand so the output for distance is different, but the outputs
+# are more accurate than the one used for the example.
+
+# In all cases distance is final_position-initial_position
+# TOTAL distance is final_position+initial_position
+# If you want to find distance from one point to another assume initial_position=0
+
+# Typing hints
+from typing import Union, Optional
+
+# FINDING ACCELERATION
+
+
+class NotEnoughInfo(Exception):
+    pass
+
+
+# retaining_wall is the rounding place
+def acceleration(
+    initial_velocity: Optional[Union[int, float]] = None,
+    final_velocity: Optional[Union[int, float]] = None,
+    final_position: Optional[Union[int, float]] = None,
+    initial_position: Optional[Union[int, float]] = None,
+    time_elapsed: Optional[Union[int, float]] = None,
+    retaining_wall: Optional[int] = None,
+) -> float:
+    """
+    Find acceleration for given initial_position, final_position, initial_velocity,
+    final_velocity, or time_elapsed.
+
+    :param initial_position: int, float
+    :param final_position: int, float
+    :param initial_velocity: int, float
+    :param final_velocity: int, float
+    :param time_elapsed: int, float
+    :param retaining_wall: int
+    :return: float
```

Comment:
This is repetitive of the info is already stated in the type hints above and modern document generators no longer need data in this format.  A description of the purpose and/or limits of each parameter or return value would be a helpful addition.